### PR TITLE
Refactoring & state-aware illustrations for Settings>BackgroundPriority (EXPOSUREAPP-3632)

### PR DESCRIPTION
**Pull request by d4rken**
Settings>BackgroundPriority  illustrations were not state aware, they always showed the same icon + color independent of the current background priority settings.

This PR adds new illustrations with matching icons (enabled/green + disabled/red).

When I wanted write logic I was looking for some kind of `backgroundPriority: Flow<Boolean>` and noticed the :spaghetti: around `ConnectivityHelper` again, which was still up for refactoring. 

I moved relevant logic into `BackgroundModeStatus`, and deleted the now superfluos code from ConnectivityHelper, see:
https://github.com/corona-warn-app/cwa-app-android/pull/1997/files#diff-7e2414bf7818e57619fd3c9606df0f476447adaac4f9fc775bf18b699a38c9a7

## Testing
* Check the dialogs that pop up on first install
* Check state changes in settings

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/backgroundpriority/BackgroundPriorityState.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import android.content.Context
+import android.graphics.drawable.Drawable
+import androidx.core.content.ContextCompat
+import de.rki.coronawarnapp.R
+
+data class BackgroundPriorityState(
+    val isBackgroundPriorityEnabled: Boolean
+) {
+    fun getButtonStateLabel(c: Context): String = c.getString(
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/backgroundpriority/BackgroundPriorityState.kt:11]: Please rename `c` to `context`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/backgroundpriority/BackgroundPriorityState.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        if (isBackgroundPriorityEnabled) R.string.settings_on
+        else R.string.settings_off
+    )
+
+    fun getHeaderIllustration(c: Context): Drawable? {
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/backgroundpriority/BackgroundPriorityState.kt:16]: Same here

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/device/PowerManagement.kt [hunk 3]
```diff
@@ -1,15 +1,32 @@
...
+class PowerManagement @Inject constructor(
+    @AppContext private val context: Context
+) {
 
-    /**
-     * Checks if app is excluded from battery optimizations
-     */
-    val isIgnoringBatteryOptimizations: Boolean
+    private val powerManager by lazy {
+        context.getSystemService(Context.POWER_SERVICE) as PowerManager
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/device/PowerManagement.kt:18]: `context.getSystemService< PowerManager>`?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/di/AndroidModule.kt [hunk 4]
```diff
@@ -50,4 +51,9 @@ class AndroidModule {
 
     @Provides
     fun navDeepLinkBuilder(@AppContext context: Context): NavDeepLinkBuilder = NavDeepLinkBuilder(context)
+
+    @Provides
+    @Singleton
+    fun activityManager(@AppContext context: Context): ActivityManager =
+        context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/di/AndroidModule.kt:58]: `context.getSystemService< ActivityManager >`

**Review by ralfgehrer — DISMISSED** [review]
LGTM. Tested on Pixel in light and dark mode

**Review by mtwalli — DISMISSED** [review]

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/risk/RiskLevelTaskTest.kt [hunk 5]
```diff
@@ -85,22 +97,140 @@ class RiskLevelTaskTest : BaseTest() {
...
+    }
+
+    @Test
+    fun `risk calculation is skipped if results are outdated while in background mode`() = runBlockingTest {
+        val cachedKey = mockk<CachedKey>().apply {
+            every { info } returns mockk<CachedKeyInfo>().apply {
+                every { toDateTime() } returns DateTime.now().minusDays(3)
+            }
+        }
+        val now = Instant.now()
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/risk/RiskLevelTaskTest.kt:172]: Minor: A fixed date would be preferable in a test case.

**Review by ralfgehrer — APPROVED** [review]

**Pull request merged**